exec/java-exec/src/main/java/org/apache/drill/exec/store/pojo/PojoRecordReader.java

```diff
@@ -47,24 +46,30 @@
 import org.apache.drill.exec.vector.AllocationHelper;
 import org.apache.drill.exec.vector.ValueVector;
 
+import com.google.common.collect.ImmutableList;
 import com.google.common.collect.Lists;
 
-public class PojoRecordReader<T> extends AbstractRecordReader {
+public class PojoRecordReader<T> extends AbstractRecordReader implements Iterable<T> {
   private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(PojoRecordReader.class);
   private static final ControlsInjector injector = ControlsInjectorFactory.getInjector(PojoRecordReader.class);
 
-  public final int forJsonIgnore = 1;
-
   private final Class<T> pojoClass;
-  private final Iterator<T> iterator;
+  private final List<T> pojoObjects;
   private PojoWriter[] writers;
   private boolean doCurrent;
   private T currentPojo;
   private OperatorContext operatorContext;
 
+  private Iterator<T> currentIterator;
+
+  /**
+   * TODO: Cleanup the callers to pass the List of POJO objects directly rather than iterator.
+   * @param pojoClass
+   * @param iterator
+   */
   public PojoRecordReader(Class<T> pojoClass, Iterator<T> iterator) {
     this.pojoClass = pojoClass;
-    this.iterator = iterator;
+    this.pojoObjects = ImmutableList.copyOf(iterator);
```

Comment:
We lose laziness by doing this. I am not sure if queries will be impacted (queries on information schema and sys tables), since this will be done at setup time?